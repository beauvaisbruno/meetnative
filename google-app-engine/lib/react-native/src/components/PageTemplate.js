"use strict";
var __importStar =
  (this && this.__importStar) ||
  function(mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null)
      for (var k in mod)
        if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
    result["default"] = mod;
    return result;
  };
var __importDefault =
  (this && this.__importDefault) ||
  function(mod) {
    return mod && mod.__esModule ? mod : { default: mod };
  };
Object.defineProperty(exports, "__esModule", { value: true });
const React = __importStar(require("react"));
const native_base_1 = require("native-base");
const HeaderApp_1 = __importDefault(require("./HeaderApp"));
const custom_1 = __importDefault(
  require("../config/native-base-theme/variables/custom")
);
function PageTemplate(Screen, headerTitle, padding = true, noScroll = false) {
  return class extends React.Component {
    render() {
      return React.createElement(
        native_base_1.Container,
        { style: { backgroundColor: custom_1.default.backgroundApp } },
        React.createElement(HeaderApp_1.default, { headerTitle: headerTitle }),
        noScroll ||
          React.createElement(
            native_base_1.Content,
            { padder: padding },
            React.createElement(Screen, Object.assign({}, this.props))
          ),
        noScroll && React.createElement(Screen, Object.assign({}, this.props))
      );
    }
  };
}
exports.default = PageTemplate;
//# sourceMappingURL=PageTemplate.js.map
