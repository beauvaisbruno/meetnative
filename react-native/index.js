require("./src/config/debugHook");

import { AppRegistry } from "react-native";
import { name as appName } from "./app.json";
import StorybookUI from "./src/config/storybook";
import App from "./src/App";

//AppRegistry.registerComponent(appName, () => StorybookUI);
AppRegistry.registerComponent(appName, () => App);
