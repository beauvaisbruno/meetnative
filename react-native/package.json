{
  "name": "meetnative",
  "version": "1.0.1",
  "private": true,
  "scripts": {
    "android reload": "adb shell input text \"RR\"",
    "android": "adb reverse tcp:8097 tcp:8097 && adb reverse tcp:9090 tcp:9090 && react-native run-android",
    "android open debug menu": "adb shell input keyevent 82",
    "android build": "cd android && gradlew assembleRelease",
    "android start release": "adb reverse tcp:8097 tcp:8097 && adb reverse tcp:9090 tcp:9090 && react-native run-android --variant=release",
    "android clean": "cd android && gradlew clean",
    "stories update loader": "rnstl --pattern ./src/**/*.stories.{ts,tsx} --outputFile ./src/config/storyLoader.js",
    "test coverage": "jest --coverage --verbose --collectCoverageFrom='[\"src/**/*.{ts,tsx}\",\"!src/**/stories/*\"]'",
    "test": "jest",
    "test CI": "jest --ci --runInBand"
  },
  "dependencies": {
    "@storybook/addon-actions": "^5.0.0",
    "@storybook/react": "^5.0.0",
    "@storybook/react-native": "^4.1.11",
    "apollo-cache-inmemory": "^1.5.1",
    "apollo-client": "^2.5.1",
    "apollo-link": "^1.2.8",
    "apollo-link-context": "^1.0.14",
    "apollo-link-error": "^1.1.7",
    "apollo-link-http": "^1.5.11",
    "apollo-link-redux": "^0.2.1",
    "apollo-link-state": "^0.4.2",
    "apollo-link-ws": "^1.0.14",
    "apollo-storybook-react-native": "^0.2.9",
    "apollo-utilities": "^1.2.1",
    "assert": "^1.4.1",
    "deepmerge": "2.2.1",
    "eslint-plugin-react": "^7.12.4",
    "formik": "^1.4.1",
    "fraql": "^1.2.1",
    "geolib": "latest",
    "graphql": "^14.1.1",
    "graphql-tag": "^2.10.1",
    "graphql-tools": "^4.0.4",
    "messageformat": "^2.0.5",
    "moment": "^2.24.0",
    "native-base": "^2.12.0",
    "ngeohash": "^0.6.3",
    "react": "16.6.3",
    "react-apollo": "^2.5.1",
    "react-native": "0.58.6",
    "react-native-auto-height-image": "^1.0.5",
    "react-native-fbsdk": "^0.8.0",
    "react-native-firebase": "^5.2.3",
    "react-native-gesture-handler": "^1.0.12",
    "react-native-google-places-autocomplete": "^1.3.9",
    "react-native-google-signin": "^1.0.2",
    "react-native-localize": "^1.0.1",
    "react-native-responsive-screen": "^1.2.0",
    "react-navigation": "^3.0.8",
    "react-redux": "^6.0.0",
    "redux": "^4.0.1",
    "redux-actions": "^2.6.4",
    "redux-devtools-extension": "^2.13.7",
    "redux-logger": "latest",
    "redux-persist": "^5.10.0",
    "redux-saga": "^0.16.2",
    "redux-saga-firebase": "^0.14.1",
    "redux-saga-rn-alert": "^1.1.5",
    "stream": "latest",
    "styled-components": "^4.1.3",
    "subscriptions-transport-ws": "^0.9.15",
    "typesafe-actions": "^3.0.0",
    "yup": "^0.26.6"
  },
  "devDependencies": {
    "@types/enzyme": "^3.1.15",
    "@types/enzyme-adapter-react-16": "^1.0.3",
    "@types/es6-promise": "^3.3.0",
    "@types/jest": "^23.3.13",
    "@types/ngeohash": "^0.6.2",
    "@types/react": "^16.7.18",
    "@types/react-native": "^0.57.27",
    "@types/react-native-fbsdk": "^0.6.2",
    "@types/react-navigation": "^3.0.1",
    "@types/react-redux": "^6.0.12",
    "@types/react-test-renderer": "^16.0.3",
    "@types/redux-actions": "^2.3.1",
    "@types/redux-mock-store": "^1.0.0",
    "@types/redux-test-utils": "^0.2.0",
    "@types/storybook__react": "^4.0.0",
    "@types/storybook__react-native": "^3.0.3",
    "@types/styled-components": "^4.1.8",
    "@types/yup": "^0.26.4",
    "apollo-cache-persist": "^0.1.1",
    "apollo-storybook-decorator": "^0.5.1",
    "babel-jest": "23.6.0",
    "enzyme": "^3.8.0",
    "enzyme-adapter-react-16": "^1.7.1",
    "enzyme-redux": "latest",
    "enzyme-to-json": "^3.3.5",
    "eslint": "^5.12.1",
    "eslint-config-react-native": "^4.0.0",
    "eslint-plugin-jest": "^22.1.3",
    "eslint-plugin-react-native": "^3.5.0",
    "immutability-helper": "^3.0.0",
    "jest": "23.6.0",
    "jest-fetch-mock": "^2.1.0",
    "jest-junit": "^6.3.0",
    "metro-react-native-babel-preset": "0.51.0",
    "prettier": "^1.15.3",
    "react-dom": "^16.7.0",
    "react-native-graphql-transformer": "^0.1.3",
    "react-native-storybook-loader": "^1.8.0",
    "react-native-typescript-transformer": "^1.2.11",
    "react-native-vector-icons": "^6.3.0",
    "react-test-renderer": "16.6.3",
    "react-testing-library": "^5.4.4",
    "reactotron-react-native": "^2.1.3",
    "reactotron-redux": "^2.1.3",
    "reactotron-redux-saga": "^2.1.3",
    "redux-mock-store": "latest",
    "redux-saga-test-plan": "latest",
    "redux-test-utils": "latest",
    "ts-jest": "^23.10.5",
    "tslint": "^5.11.0",
    "tslint-config-blvd": "^1.2.1",
    "typescript": "^3.2.2",
    "typescript-eslint-parser": "^21.0.2",
    "waait": "^1.0.4"
  },
  "jest": {
    "preset": "react-native",
    "transform": {
      "^.+\\.js$": "<rootDir>/node_modules/react-native/jest/preprocessor.js",
      "^.+\\.tsx?$": "ts-jest"
    },
    "transformIgnorePatterns": [
      "/node_modules/(?!native-base)/"
    ],
    "globals": {
      "ts-jest": {
        "tsConfig": "jest/tsconfig.jest.json"
      }
    },
    "testRegex": "(/__tests__/.*|(\\.|/)(test|spec))\\.(tsx?)$",
    "moduleFileExtensions": [
      "ts",
      "tsx",
      "js",
      "jsx",
      "json",
      "node"
    ],
    "modulePaths": [
      "<rootDir>"
    ],
    "setupTestFrameworkScriptFile": "./jest/jest.before.js",
    "setupFiles": [
      "./jest/jest.setup.js"
    ],
    "collectCoverageFrom": [
      "src/**/*.{ts,tsx}",
      "!src/**/*stories*"
    ]
  }
}
