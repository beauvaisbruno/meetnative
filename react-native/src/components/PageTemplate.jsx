"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
var React = require("react");
var native_base_1 = require("native-base");
var HeaderApp_1 = require("./HeaderApp");
var custom_1 = require("../config/native-base-theme/variables/custom");
function PageTemplate(Screen, headerTitle, padding, noScroll) {
    if (padding === void 0) { padding = true; }
    if (noScroll === void 0) { noScroll = false; }
    return /** @class */ (function (_super) {
        __extends(class_1, _super);
        function class_1() {
            return _super !== null && _super.apply(this, arguments) || this;
        }
        class_1.prototype.render = function () {
            return (<native_base_1.Container style={{ backgroundColor: custom_1.default.backgroundApp }}>
          <HeaderApp_1.default headerTitle={headerTitle}/>
          {noScroll || (<native_base_1.Content padder={padding}>
              <Screen {...this.props}/>
            </native_base_1.Content>)}
          {noScroll && <Screen {...this.props}/>}
        </native_base_1.Container>);
        };
        return class_1;
    }(React.Component));
}
exports.default = PageTemplate;
