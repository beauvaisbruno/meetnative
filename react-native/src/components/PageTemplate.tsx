import * as React from "react";
import {
  Body,
  Button,
  Container,
  Content,
  Header,
  Icon,
  Left,
  Right,
  Title
} from "native-base";
import { NavigationInjectedProps } from "react-navigation";
import HeaderApp from "./HeaderApp";
import custom from "../config/native-base-theme/variables/custom";

function PageTemplate(
  Screen: any,
  headerTitle: string,
  padding = true,
  noScroll = false
) {
  return class extends React.Component<NavigationInjectedProps> {
    render() {
      return (
        <Container style={{ backgroundColor: custom.backgroundApp }}>
          <HeaderApp headerTitle={headerTitle} />
          {noScroll || (
            <Content padder={padding}>
              <Screen {...this.props} />
            </Content>
          )}
          {noScroll && <Screen {...this.props} />}
        </Container>
      );
    }
  };
}

export default PageTemplate;
