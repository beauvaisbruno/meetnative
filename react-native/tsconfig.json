{
  "include": [
    "src/**/*"
  ],
  "exclude": [
    "node_modules/**/*"
  ],
  "compilerOptions": {
    "target": "es2015",                          /* Specify ECMAScript target version: 'ES3' (default), 'ES5', 'ES2015', 'ES2016', 'ES2017','ES2018' or 'ESNEXT'. */
    "module": "commonjs",                     /* Specify module code generation: 'none', 'commonjs', 'amd', 'system', 'umd', 'es2015', or 'ESNext'. */
    "jsx": "react",                           /* Specify JSX code generation: 'preserve', 'react-native', or 'react'. */
    "strict": true,                           /* Enable all strict type-checking options. */
    "noImplicitAny": true,                 /* Raise error on expressions and declarations with an implied 'any' type. */
    "allowSyntheticDefaultImports": true,  /* Allow default imports from modules with no default export. This does not affect code emit, just typechecking. */
    "esModuleInterop": true
  }
}
