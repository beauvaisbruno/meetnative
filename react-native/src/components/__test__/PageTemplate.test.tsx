import { shallow } from "enzyme";
import * as React from "react";
import PageTemplate from "../PageTemplate";
import { View } from "react-native";
import toJson from "enzyme-to-json";

it("pageTemplate", () => {
  class Comp extends React.Component<any> {
    render() {
      return <View />;
    }
  }
  const openDrawer = jest.fn();
  const Node = PageTemplate(Comp, "Home");
  // @ts-ignore
  const tree = shallow(<Node navigation={{ openDrawer }} />);
  expect(toJson(tree)).toMatchSnapshot();
});
