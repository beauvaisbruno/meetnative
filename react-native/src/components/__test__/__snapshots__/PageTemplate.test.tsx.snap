// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pageTemplate 1`] = `
<Styled(Container)
  style={
    Object {
      "backgroundColor": "white",
    }
  }
>
  <Connect(withNavigation(HeaderApp))
    headerTitle="Home"
  />
  <Styled(Content)
    padder={true}
  >
    <Comp
      navigation={
        Object {
          "openDrawer": [MockFunction],
        }
      }
    />
  </Styled(Content)>
</Styled(Container)>
`;
